### PR TITLE
adds private app type and new fields

### PR DESCRIPTION
**Pull request by AkemiDavisson**
## Description

Adds private app type and related fields

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: access_application.go [hunk 1]
```diff
@@ -44,7 +44,13 @@ type AccessApplication struct {
 	SameSiteCookieAttribute string                        `json:"same_site_cookie_attribute,omitempty"`
 	LogoURL                 string                        `json:"logo_url,omitempty"`
 	SkipInterstitial        bool                          `json:"skip_interstitial,omitempty"`
-	AppLauncherVisible    bool                            `json:"app_launcher_visible,omitempty"`
+	AppLauncherVisible      bool                           `json:"app_launcher_visible,omitempty"`
```
Comment (jacobbednarz) [inline access_application.go:47]: can you run `go fmt` on this file? CI should have failed on this but didn’t :/

File: access_application_test.go [hunk 2]
```diff
@@ -455,3 +455,79 @@ func TestAccessApplicationWithCORS(t *testing.T) {
...
+	}
+
+	createdAt, _ := time.Parse(time.RFC3339, "2014-01-01T05:20:00.12345Z")
+	updatedAt, _ := time.Parse(time.RFC3339, "2014-01-01T05:20:00.12345Z")
+	fullAccessApplication := AccessApplication{
+		ID:                     "480f4f69-1a28-4fdd-9240-1ed29f0ac1db",
+		Name:                   "Private Admin Site",
+		PrivateAddress:         "198.51.100.0",
+		Type:                   "private_ip",
+		GatewayRules:   			  []AccessApplicationGatewayRule{
```
Comment (jacobbednarz) [inline access_application_test.go:503]: same here with `go for‘

File: access_application_test.go [hunk 3]
```diff
@@ -455,3 +455,79 @@ func TestAccessApplicationWithCORS(t *testing.T) {
...
+				"type": "private_ip",
+		    "gateway_rules": [
+					{"id": "d9c61460-6f4d-4c40-89ea-2f552c9a8466"},
+					{"id": "bc5ee7e7-9773-47a3-835e-b9b9799ebb92"}
+				],
+				"session_duration": "24h",
+				"allowed_idps": ["f174e90a-fafe-4643-bbbc-4a0ed4fc8415"],
+				"auto_redirect_to_identity": false,
+				"enable_binding_cookie": false,
+				"custom_deny_url": "https://www.cloudflare.com",
```
Comment (jacobbednarz) [inline access_application_test.go:486]: where we are using a URL in tests, can you please swap to `example.com`?

Comment (AkemiDavisson) [reply access_application_test.go]: Updated!

File: access_application.go [hunk 4]
```diff
@@ -26,33 +26,39 @@ const (
...
+	ID string `json:"id,omitempty"`
 }
 
 // AccessApplicationCorsHeaders represents the CORS HTTP headers for an Access
 // Application.
 type AccessApplicationCorsHeaders struct {
 	AllowedMethods   []string `json:"allowed_methods,omitempty"`
 	AllowedOrigins   []string `json:"allowed_origins,omitempty"`
-	AllowedHeaders   []string `json:"allowed_headers,omitempty"`
+	AllowedHeaders   []string `json:"gallowed_headers,omitempty"`
```
Comment (jacobbednarz) [inline access_application.go:61]: i'm guessing this needs to be

```suggestion
	AllowedHeaders   []string `json:"allowed_headers,omitempty"`
```

**Comment by jacobbednarz** [comment]
thanks @AkemiDavisson 🍭

**Pull request merged**